app/lib/features/chat_ng/widgets/chat_editor/chat_editor.dart

```diff
@@ -102,21 +102,8 @@ class _ChatEditorState extends ConsumerState<ChatEditor> {
 
     final body = msgContent.body();
     if (body.isEmpty) return;
-    // clear the editor first
-    textEditorState.clear();
 
-    final docNode = textEditorState.getNodeAtPath([0]);
-    if (docNode == null) return;
-
-    // process text and apply mention attributes , if any
-    textEditorState.toMentionPills(body, docNode);
-
-    final text = docNode.delta?.toPlainText() ?? '';
-    final pos = Position(path: [0], offset: text.length);
-    textEditorState.updateSelectionWithReason(
-      Selection.collapsed(pos),
-      reason: SelectionUpdateReason.uiEvent,
-    );
+    textEditorState.replaceContent(body, msgContent.formattedBody());
```

Comment:
does it also takes care of non-appflowy html, which is somewhat different. The decoder fails at parsing html from Element because of few difference of tags. Which is why I normalized it prior of decoding in my [PR](https://github.com/acterglobal/a3/pull/3056/files#r2150539583)